components/script/stylesheet_loader.rs

```diff
@@ -147,6 +150,9 @@ impl FetchResponseListener for StylesheetContext {
                 }
                 StylesheetContextSource::Import(ref import) => {
                     let import = import.read();
+                    if elem.downcast::<HTMLLinkElement>().unwrap().is_alternate() {
```

Comment:
I suspect this will break if we have `<style>@import 'foo.css'</style>`, since the element is a `HTMLStyleElement`. Try running `./mach test-wpt tests/wpt/mozilla/tests/mozilla/restyle-out-of-document.html` to verify this.